packages/x-data-grid/src/utils/createSelector.ts

```diff
@@ -253,55 +145,9 @@ export const createSelectorV8 = ((
   selector.acceptsApiRef = true;
 
   return selector;
-}) as unknown as CreateSelectorFunctionV8;
+}) as unknown as CreateSelectorFunction;
 
-// TODO v8: Remove this function
 export const createSelectorMemoized: CreateSelectorFunction = (...args: any) => {
-  const selector = (stateOrApiRef: any, instanceId?: any) => {
-    const isAPIRef = checkIsAPIRef(stateOrApiRef);
-    const cacheKey = isAPIRef
-      ? stateOrApiRef.current.instanceId
-      : (instanceId ?? DEFAULT_INSTANCE_ID);
-    const state = isAPIRef ? stateOrApiRef.current.state : stateOrApiRef;
-
-    if (process.env.NODE_ENV !== 'production') {
-      if (cacheKey.id === 'default') {
-        warnOnce([
-          'MUI X: A selector was called without passing the instance ID, which may impact the performance of the grid.',
```

Comment:
tests are giving this warning, is that a typescript issue with overload call signatures of the selector method?

Reply:
@KenanYusuf owns this feature so he could give you more information. Note that we're already actively working on that issue so I'm not sure if this is a good issue for you to take, Kenan can confirm if he's open to splitting that work.

Reply:
Okay I thought it is open for anyone to take
 @KenanYusuf are you willing to collaborate for adding this new selector task in list view? if yes we can proceed and if 
  not feel free to close this PR.

Reply:
@k-rajat19 thanks for your willingness to help on this issue, however, we are still discussing the implementation of this feature internally and it is not yet ready for contributions. 

In future, if you are unsure whether an issue is ready to be picked up, feel free to ask on the issue to avoid wasted effort. Your contributions are appreciated.

Reply:
Sure, thank you
I have learned few things about the grid project while working on this. so it is somewhat worth it for me atleast😅